### PR TITLE
Implemented unit tests over GreetingService

### PR DESCRIPTION
**Pull request by ioan994**
Added Junit dependency
Added Idea and target directories to gitignore

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -22,6 +22,11 @@
             <artifactId>jersey-client</artifactId>
             <version>2.25</version>
         </dependency>
+        <dependency>
+            <groupId>junit</groupId>
+            <artifactId>junit</artifactId>
+            <version>4.12</version>
```
Comment (fland) [inline pom.xml:28]: you might use maven <scope>test</scope> https://maven.apache.org/guides/introduction/introduction-to-dependency-mechanism.html#Dependency_Scope

File: src/test/java/org/noteskeeper/server/GreetingServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+import org.junit.Test;
+
+public class GreetingServiceTest {
+
+    private GreetingService service = new GreetingService();
+
+    @Test
+    public void greetGreetsAnonymousWhenPassedNullOrEmptyName() {
+        String expectedGreeting = "Hello, Anonymous";
+        assertEquals("",expectedGreeting, service.greet(null));
```
Comment (fland) [inline src/test/java/org/noteskeeper/server/GreetingServiceTest.java]: you might omit empty assert message => assertEquals(expectedGreeting, service.great(null))

File: src/test/java/org/noteskeeper/server/GreetingServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+        assertEquals("",expectedGreeting, service.greet(""));
+    }
+
+    @Test
+    public void greetGreetsUserWhenPassedName() {
+        String expectedGreeting = "Hello, Ivan";
+        assertEquals("",expectedGreeting, service.greet("Ivan"));
```
Comment (fland) [inline src/test/java/org/noteskeeper/server/GreetingServiceTest.java]: use Ctrl+Alt+L in Idea for code formatting

**Review by fland — CHANGES_REQUESTED** [review]

**Review by fland — APPROVED** [review]

File: pom.xml (hunk 1 above)
Comment (mnovoseltsev) [reply pom.xml:28]: Hi Guys, JUnit 5 is up and running. No sense to use 4th, http://junit.org/junit5/.

Comment (fland) [reply pom.xml:28]: there's no GA (general available) or RELEASE version yet.

Comment (mnovoseltsev) [reply pom.xml:28]: You are not writing enterprise project, why do not use latest Junit as well as try java 9 ?

**Pull request merged**